dcos_launch/terraform.py

```diff
@@ -0,0 +1,285 @@
+        if m:
+            # remove prefix
+            s = m.group(0)[len(prefix):]
+            # remove whitespace
+            s = "".join(s.split())
+            # remove brackets
+            s = s[1:-1]
+            ips = s.split(',')
+    return ips
+
+
+def _convert_to_describe_format(ips: list) -> list:
+    # TODO private ips not currently outputted in terraform-dcos
+    return [{'private_ip': None, 'public_ip': ip} for ip in ips]
+
+
+class TerraformLauncher(util.AbstractLauncher):
+    def __init__(self, config: dict, env=None):
+        if env:
+            os.environ.update(env)
+        self.config = config
+        self.init_dir = dcos_launch.config.expand_path('', self.config['init_dir'])
+        self.cluster_profile_path = os.path.join(self.init_dir, 'desired_cluster_profile.tfvars')
+        self.dcos_launch_root_dir = os.path.abspath(os.path.join(self.init_dir, '..'))
+        self.terraform_binary = os.path.join(self.dcos_launch_root_dir, 'terraform')
+        self.default_priv_key_path = os.path.join(self.init_dir, 'key.pem')
+
+    def terraform_cmd(self):
+        """ Returns the right Terraform invocation command depending on whether it was installed by the user or by
+        dcos-launch.
+        """
+        binary = self.terraform_binary
+        if not os.path.exists(binary):
+            binary = 'terraform'
+        return binary
+
+    def create(self, info_path='cluster_info.json'):
+        try:
+            if os.path.exists(self.init_dir):
+                raise util.LauncherError('ClusterAlreadyExists', "Either the cluster you are trying to create is "
+                                                                 "already running or the init_dir you specified in your"
+                                                                 " config is already used by another active cluster.")
+            os.makedirs(self.init_dir)
+            # Check if Terraform is installed by running 'terraform version'. If that fails, install Terraform.
+            try:
+                subprocess.run([self.terraform_cmd(), 'version'], check=True, stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT)
+            except FileNotFoundError:
+                log.info('No Terraform installation detected. Terraform is now being installed.')
+                self._install_terraform()
+
+            if self.config['key_helper']:
+                self.key_helper()
+            else:
+                log.warning('WARNING: {}Since you did not set "key_helper: true" in your config, make sure your '
+                            'ssh-agent is running i.e. "eval `ssh-agent -s`" and that you have added your private key '
+                            'to it i.e. "ssh-add /path/to/key.pem". ssh-agent usage is specific to terraform, not '
+                            'dcos-launch.'.
+                            format('\033[93m'))  # represents the yellow color
```

Comment:
Please drop the color bits. This might cause more trouble than it is worth (might not show up as intended in all terminals). If we did want to do log coloring, I would push for doing it with a module as opposed to hard coding bytes in a format string, but that is highly tangential to this PR